src/comm/TcpAcceptor.cc

```diff
@@ -375,8 +380,6 @@ Comm::TcpAcceptor::oldAccept(Comm::ConnectionPointer &details)
     // so we end up with a uniform "(HTTP|FTP-data|HTTPS|...) remote-ip:remote-port"
     fd_open(sock, FD_SOCKET, "HTTP Request");
     details->fd = sock;
-    details->enterOrphanage();
```

Comment:
IMO this (removed) call was placed correctly - the Connection was considered as 'orphan' since the 'details' variable (created and passed from acceptOne()) is not stored in TcpAcceptor itself. In other words, if an exception occurs and passes beyond acceptOne() and oldAccept() the 'details' will be lost.

Reply:
Going by Alex's definition of orphans as "expected" vs "unexpected" errors the move from here to notify() is correct.

"expected" errors and exceptions are caught and handled before leading to notify() which should perform the closure or orphanage. All others are "unexpected" and should not be silenced by the orphanage flag being set this early.

Reply:
> Eduard: IMO this (removed) call was placed correctly - the Connection was considered as 'orphan' since the 'details' variable (created and passed from acceptOne()) is not stored in TcpAcceptor itself.

Yes, `details` is a correctly handled orphan here and, hence, the enterOrphanage() call should not be moved lower. If anything, the call can be moved _higher_ but that would require enlarging my definition of an orphan Connections to also include not-opened connections (and that definition adjustment is difficult to make correctly because it requires defining Connection ownership).

> Amos: Going by Alex's definition of orphans as "expected" vs "unexpected" errors the move from here to notify() is correct.

I do not know what "orphan" definition you are using, but there is nothing about expected or unexpected errors in my definition from https://github.com/squid-cache/squid/pull/837#discussion_r645998210 (quoted below):

> Technically, an orphan Connection object is a Connection object that gets destructed in an open state (i.e. with a non-negative descriptor). Some _rare_ orphan Connection cases are not covered by this statement, but I am omitting them here for simplicity sake. On a more human level, an orphan Connection object is a Connection object without an owner (where Connection owner is code that explicitly closes the connection or has an associated connection closure handler). 

We should call Connection::enterOrphanage() for x when the following two conditions are both true:

* Connection x is an orphan (see my definition quoted above). For example, placing a "throw 42;" statement immediately after the enterOrphanage() call would result in x destruction while x->fd is not negative.
* We are OK with Connection x being destructed in an open state. Such destruction and the associated fd auto-closure is not a bug (for this x) but a desirable outcome.

The specific enterOrphanage() call in question appears to satisfy all these criteria. I see no reason to (re)move it.

> "expected" errors and exceptions are caught and handled before leading to notify() which should perform the closure or orphanage.

That assertion looks false to me as far as exceptions are concerned: Plenty of function calls right below the existing enterOrphanage() call may throw an exception (now or in the future code). Those exceptions are not caught by any code that can close the open Connection in question (that is why that Connection object is an orphan!). And none of those exceptions would lead to notify() with this Connection object as a parameter.

The existing enterOrphanage() call essentially says: If the code below throws, please automatically close this Connection without complaining -- it is the best thing to do to this Connection under those circumstances. I see no problems with that statement. Do you?

Reply:
> yadij added S-waiting-for-reviewer and removed S-waiting-for-author labels

@yadij, I think resolving this change request (and my related higher level concerns expressed earlier) is still on your side so I am relabeling this with S-waiting-for-author. Please correct me if I misinterpreted the PR state.

Reply:
Update: when the Comm::Descriptor proposal is merged I will update this PR to be just a naming fix of s/conn/listenConn/

Reply:
> when the Comm::Descriptor proposal is merged I will update this PR to be just a naming fix of s/conn/listenConn/

Understood. I labeled the PR as `S-waiting-for-PR` while we wait for #1103.

FWIW, I do not think listenConn is a good new name because the object does not represent a connection (we are simply abusing a Connection class for something that does not represent a connection or connection end point). Something like `listeningSocket_`, `'listener_`, or even `listen_` would be better IMO.